spyder/plugins/onlinehelp/widgets.py

```diff
@@ -78,123 +101,384 @@ def completer(self):
-        elif self.server.is_running():
-            self.quit_server()
+        elif self.server.isRunning():
+            self.server.server_started.disconnect(
+                self._continue_initialization)
+            self.server.quit()
+
         self.server = PydocServer(port=self.port)
-        self.server.server_started.connect(self.initialize_continued)
+        self.server.server_started.connect(
+            self._continue_initialization)
         self.server.start()
 
     def quit_server(self):
-        """Quit the server"""
+        """Quit the server."""
         if self.server is None:
             return
+
         if self.server.is_running():
-            self.server.server_started.disconnect(self.initialize_continued)
+            self.server.server_started.disconnect(
+                self._continue_initialization)
             self.server.quit_server()
             self.server.quit()
 
-    # ------ WebBrowser API ---------------------------------------------------
     def get_label(self):
         """Return address label text"""
         return _("Package:")
 
     def reload(self):
-        """Reload page"""
+        """Reload page."""
         self.start_server()
         self.webview.reload()
 
-    def load_finished(self, ok):
-        """Handle load finished."""
-        pass
-
     def text_to_url(self, text):
-        """Convert text address into QUrl object"""
+        """
+        Convert text address into QUrl object.
+
+        Parameters
+        ----------
+        text: str
+            Url address.
+        """
         if text != 'about:blank':
             text += '.html'
+
         if text.startswith('/'):
             text = text[1:]
 
         return QUrl(self.home_url.toString() + text)
 
     def url_to_text(self, url):
-        """Convert QUrl object to displayed text in combo box"""
+        """
+        Convert QUrl object to displayed text in combo box.
+
+        Parameters
+        ----------
+        url: QUrl
+            Url address.
+        """
         string_url = url.toString()
         if 'about:blank' in string_url:
             return 'about:blank'
         elif 'get?key=' in string_url or 'search?key=' in string_url:
-            return ''
-        return osp.splitext(to_text_string(url.path()))[0][1:]
+            return url.toString().split('=')[-1]
+
+        return os.path.splitext(str(url.path()))[0][1:]
+
+    def set_home_url(self, text):
+        """
+        Set home URL.
+
+        Parameters
+        ----------
+        text: str
+            Home url address.
+        """
+        self.home_url = QUrl(text)
+
+    def set_url(self, url):
+        """
+        Set current URL.
+
+        Parameters
+        ----------
+        url: QUrl or str
+            Url address.
+        """
+        self._change_url(url)
+        self.go_to(url)
+
+    def go_to(self, url_or_text):
+        """
+        Go to page *address*.

```

Comment:
Please change this to

    Go to page URL.

Reply:
Fixed